src/D2L.CodeStyle.Annotations/Contract/OnlyVisibleToTypeAttribute.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+
+namespace D2L.CodeStyle.Annotations.Contract {
+
+	[AttributeUsage(
+		validOn: AttributeTargets.Method | AttributeTargets.Property,
+		AllowMultiple = true,
+		Inherited = false
+	)]
+	public sealed class OnlyVisibleToTypeAttribute : Attribute {
+
+		public OnlyVisibleToTypeAttribute( Type type )
+			: this( type.FullName, type.Assembly.GetName().Name ) {

```

Comment:
`type.FullName` includes type information if `IsConstructedGenericType == true`, so FYI that this might not be providing the expected value.

Is having the properties even worthwhile? Do we imagine a case for inspecting these at runtime?

Reply:
I'm happy enough to omit them.  But this only happens if you do respect at runtime.  If that's the case, I suggest we add a `ConditionalAttribute` like `SuppressMessageAttribute`.  That way the data is there if we want opt it, but not in the DLL otherwise:

https://source.dot.net/#System.Private.CoreLib/SuppressMessageAttribute.cs,22